File: app/src/androidTest/java/com/gatheringhallstudios/mhworlddatabase/dao/MonsterDaoTest.java
package com.gatheringhallstudios.mhworlddatabase.dao;

import android.content.Context;
import android.support.test.InstrumentationRegistry;
import android.support.test.runner.AndroidJUnit4;

import com.gatheringhallstudios.mhworlddatabase.data.MHWDatabase;
import com.gatheringhallstudios.mhworlddatabase.data.dao.MonsterDao;
import com.gatheringhallstudios.mhworlddatabase.data.views.MonsterView;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.List;

import static com.gatheringhallstudios.mhworlddatabase.TestUtils.getValue;
import static org.junit.Assert.assertFalse;

@RunWith(AndroidJUnit4.class)
public class MonsterDaoTest {
    private static MHWDatabase db;
    private static MonsterDao dao;

    @BeforeClass
    public static void initDatabase() {
        // this is read only, so its ok to use the actual database
        Context ctx = InstrumentationRegistry.getTargetContext();
        db = MHWDatabase.getDatabase(ctx);
        dao = db.monsterDao();
    }

    @AfterClass
    public static void closeDatabase() {
        db.close();
    }

    @Test
    public void Can_Query_MonsterList() throws Exception {
        List<MonsterView> results = getValue(dao.loadList("en"));
        assertFalse("expected results", results.isEmpty());
    }
}

File: app/src/androidTest/java/com/gatheringhallstudios/mhworlddatabase/dao/MonsterDaoTest.kt
package com.gatheringhallstudios.mhworlddatabase.dao

import androidx.test.ext.junit.runners.AndroidJUnit4

import com.gatheringhallstudios.mhworlddatabase.data.MHWDatabase
import com.gatheringhallstudios.mhworlddatabase.data.dao.MonsterDao

import org.junit.AfterClass
import org.junit.BeforeClass
import org.junit.Test
import org.junit.runner.RunWith

import com.gatheringhallstudios.mhworlddatabase.data.types.MonsterSize
import com.gatheringhallstudios.mhworlddatabase.getResult
import com.gatheringhallstudios.mhworlddatabase.initMHWDatabase
import org.junit.Assert.assertFalse
import org.junit.Assert.assertNotNull

@RunWith(AndroidJUnit4::class)
class MonsterDaoTest {

    companion object {
        private lateinit var db: MHWDatabase
        private lateinit var dao: MonsterDao

        @BeforeClass @JvmStatic
        fun initDatabase() {
            // this is read only, so its ok to use the actual database
            db = initMHWDatabase()
            dao = db.monsterDao()
        }

        @AfterClass @JvmStatic
        fun closeDatabase() {
            db.close()
        }
    }

    @Test
    fun Can_Query_MonsterList() {
        val results = dao.loadMonsters("en").getResult()
        assertFalse("expected results", results.isEmpty())
    }

    @Test
    fun Can_Query_MonsterBreaks() {
        val largeMonsters = dao.loadMonsters("en", MonsterSize.LARGE).getResult()
        val monster = largeMonsters.find { it.name == "Rathalos" }
        assertNotNull("Expected Rathalos to exist", monster)

        val breaks = dao.loadBreaks("en", monster!!.id).getResult()
        assertFalse("Expected results", breaks.isEmpty())
    }


}
